src/components/QuestEditorAddUpdate.vue

```diff
@@ -50,180 +50,187 @@
+    props: {
+        questToEdit: Object as PropType<Quest>,
+        editMode: {
+            type: String as PropType<EditorMode>,
+            required: true,
+        },
+    },
+
+    setup(props, { emit }) {
+        let behavior: EditorBehavior|null = null;
+        const { authToken } = useAuthTokenInheritor();
+        const formInputs: FormInputs = {
+            nameInput: ref(null),
+            visibilityCheck: ref(null),
+            typeSelect: ref(null),
+            regionInput: ref(null),
+            descriptionInput: ref(null),
+            objectiveTitleInput: ref(null),
+        };
+        let objectivesList: Ref<Array<Objective>> = ref([]);
+
+        const editorTitle = computed(() => {
+            switch (props.editMode) {
                 case EditorMode.ADD:
                     return "New Quest";
                 case EditorMode.UPDATE:
                     return "Update Quest";
                 default:
                     return "ERROR";
             }
-        }
-
-        get inEditMode(): boolean {
-            return this.editMode === EditorMode.UPDATE;
-        }
+        });
+        const editorMode = ref(EditorMode);
 
         /**
          * Fires an event to request going back to the list page without attempting to save anything
          */
-        backToListNoSave() {
-            this.$emit("editor-exit");
+        function backToListNoSave() {
+            emit("editor-exit");
         }
-
         /**
          * Attempts to save changes to the backend then fires an event to request a return to the list page
          */
-        async backToListAndSave() {
-            const formInputs = this.formInputs;
+        async function backToListAndSave() {
             const behaviorInputPromises: Promise<any>[] = [];
 
-            behaviorInputPromises.push(this.behavior.setName(formInputs.nameInput.value));
-            behaviorInputPromises.push(this.behavior.setSourceRegion(formInputs.regionInput.value));
-            behaviorInputPromises.push(this.behavior.setDescription(formInputs.descriptionInput.value));
-            this.objectivesList.forEach((objective) => {
+            if (behavior === null) {
+                throw new Error('Behavior was never configured!');
+            }
+
+            behaviorInputPromises.push(behavior.setName(formInputs.nameInput.value?.value ?? "No Name"));
+            behaviorInputPromises.push(behavior.setSourceRegion(formInputs.regionInput.value?.value ?? "No Region"));
+            behaviorInputPromises.push(behavior.setDescription(formInputs.descriptionInput.value?.value ?? "No Description"));
+            objectivesList.value.forEach((objective) => {
                 if (objective.text.length > 0) {
-                    behaviorInputPromises.push(this.behavior.updateObjectiveText(objective.id, objective.text))
+                    behaviorInputPromises.push(behavior?.updateObjectiveText(objective.id, objective.text) ?? Promise.resolve())
```

Comment:
Make a local copy of behavior in this function so the safe call on line 128 isn't necessary